GPT/beta-commands/gpt-function-calling.py

```diff
@@ -0,0 +1,101 @@
+from talon import Module, actions, settings
+from typing import Callable
+import requests, os, json
+from ..gpt import notify
+from .gpt_callables import function_specs, search_for_command, notify_user, display_response
+
+mod = Module()
+
+def gpt_function_query(prompt: str, content: str, insert_response: Callable[[str], str]) -> str:
+    notify("GPT Task Started")
+
+    try:
+        TOKEN = os.environ["OPENAI_API_KEY"]
+    except:
+        notify("GPT Failure: env var OPENAI_API_KEY is not set.")
+        raise Exception("GPT Failure")
+
+    url = "https://api.openai.com/v1/chat/completions"
+    headers = {
+        "Content-Type": "application/json",
+        "Authorization": f"Bearer {TOKEN}",
+    }
+    data = {
+        "messages": [
+            {
+                "role": "system",
+                "content": f"language:\n{actions.code.language()}",
+            },
+            {"role": "user", "content": f"{prompt}:\n{content}"},
+        ],
+        "tools": function_specs,
+        "max_tokens": 2024,
+        "temperature": 0.6,
+        "n": 1,
+        "stop": None,
+        "model": settings.get("user.openai_model"),
```

Comment:
Eventually will want to extract this to talon settings. Doesn't necessarily have to be in this PR though.